### ISSUE TITLE
[Feature]

### ISSUE THREAD

**Issue by idd79** [philc/vimium#2427]
I really like Chrome and Vimium. However right now I am using Firefox due to Vimfx. One of the reasons is that I use visual mode quite frequently and I find the implementation of caret/visual mode much easier in vimfx than in vimium.

By hitting 'v', Vimfx first displays hints just like when you hit 'f', which allows to you go right away to the sentence, box, or sometimes word that you want to highlight in visual mode, which I find it extremely faster and convenient over searching for a keyword first and then enter visual mode in vimium.

It woud be great if you could implement this feature. Thanks in advance.

**Comment by tlegutko** [comment]
yes I'd also love that! when choosing between Vimium and [surfingkeys](https://chrome.google.com/webstore/detail/surfingkeys/gfbliohnnapiefjpjlpjnehglfpaknnc) that was the thing I miss the most in Vimium. when pressing `v` one gets similar behavior to `f` (a'la ace-jump / easyMotion), but with hints in different color (grey). when option is selected it enters caret mode and `v` has to be pressed again for visual. 

it's sooo convenient, selecting text for copying in Vimium is one of very few things I still find more convenient to do by cursor (searching often doesn't get me close enough to the region I want). I tried surfingKeys and cVim, but I like Vimium the most and I'd love more convenient visual mode entering.

**Comment by snippins** [comment]
This is such a killer feature to have.

**Comment by alon91** [comment]
This is a really old feature request, but something I've wanted for **years**. Cool to find a bug for it.

@tlegutko: thank you sooo much for the [Surfingkeys](https://chrome.google.com/webstore/detail/surfingkeys/gfbliohnnapiefjpjlpjnehglfpaknnc) link. I strongly doubt I'd leave Vimium because switching costs would be so high, but it's super neat to see Visual Mode executed so well elsewhere. Sharing a screenshot for what that the experience looks like -- I would be so happy if this was implemented here.

In meanwhile the suggestion in [this](https://github.com/philc/vimium/issues/3179#issuecomment-449916522) thread is worth experimenting with, although definitely not as good as the Surfingkeys solution 😇

![image](https://user-images.githubusercontent.com/35006176/67630666-7bcee080-f848-11e9-8605-1b393d6346ac.png)

**Comment by random-un** [comment]
Thought I'd bump this thread 😇 

Vimium remains an indispensable productivity tool. But I find that for me, selecting and yanking text continues to require mouse usage. I never got the hang of `v` + `/` as I found it to be slow and inaccurate.

Is there a way to advocate for the development of this feature request? Anything a non-developer can help with (writing requirements, etc)? As noted in this thread, Vimfx and Surfingkeys have had it for quite some time, and bringing this functionality to Vimium would solve a decades-old problem for me!

cc'ing @smblott-github and @gdh1995 as they have previously responded to related feature requests. Many thanks in advance!!

**Comment by ivanddm** [comment]
@random-un Take a look at [vimium-c](https://github.com/gdh1995/vimium-c). It is derived from Vimium but the maintainer has added additional features including this one. I have been using it for quite awhile now.

**Comment by gdh1995** [comment]
Yes Vimium C has a similar feature request (https://github.com/gdh1995/vimium-c/issues/3) and I've added `LinkHints.activateModeToSelect` months ago.

**Comment by sisrfeng** [comment]
I think `map v LinkHints.activateModeToSelect` should be set as default

**Comment by random-un** [comment]
> Yes, [Vimium C](https://github.com/gdh1995/vimium-c) has a similar feature request ([gdh1995/vimium-c#3](https://github.com/gdh1995/vimium-c/issues/3)) and I've added `LinkHints.activateModeToSelect` months ago.

This feature is absolutely outstanding, and has led me to migrate over to Vimium-C over the last year or so. I use this command many times each day, and find the non-hints implementation impossible to use effectively.

Would the Vimium community consider adding this feature? I work for a company that severely limits the extensions that we can use due to permission concerns. Vimium was recently banned, but we do have an internal-only fork that is actively updated and used widely throughout the company.

I worry that Vimium-C will one day be removed from my corporate machine due to permissions issues and that I'll lose the ability to `LinkHints.activateModeToSelect` (now shortened to `enterVisualLineMode`). I think many Vimium users would benefit greatly from the wonderful implementation created by @gdh1995.

**Comment by gdh1995** [comment]
Sorry but I'm not interested in adding new features to this Vimium - there're too many things to fix and add in my Vimium C .

**Comment by ParkerHGordon** [comment]
This is the main thing I'm missing from SurfingKeys as well, but SurfingKeys doesn't detect the elements on exactly one of the pages I need to use regularly.
